app/process_proposal.go

```diff
@@ -103,7 +103,7 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 				}
 			}
 
-			commitment, err := inclusion.GetCommit(cacher, dah, int(malleatedTx.ShareIndex), shares.BlobSharesUsed(int(pfb.BlobSize)))
+			commitment, err := inclusion.GetCommit(cacher, dah, int(malleatedTx.ShareIndex), shares.SparseSharesNeeded(uint32(pfb.BlobSize)))
```

Comment:
FWIW it seems very hacky to convert a `uint64` => ` uint32` like this because overflows. We should consider changing https://github.com/rootulp/celestia-app/blob/e049c4eb97c5935728b7bf8fe6f63a42706a6772/proto/blob/tx.proto#L30 to `uint32` or being more defensive here